etna/search/views.py

```diff
@@ -83,21 +92,11 @@ def get_buckets(
             for bucket in bucket_list:
                 bucket.result_count = doc_counts_by_key.get(bucket.key, 0)
 
-        if current_bucket_key:
-            # set 'is_current=True' for the relevant bucket
-            for bucket in bucket_list:
-                if bucket.key == current_bucket_key:
-                    bucket.is_current = True
-                    break
-
```

Comment:
To get the current bucket, there is `BucketList.current` have used it over [here ](https://github.com/nationalarchives/ds-wagtail/blob/d71188a56a39682c24a816feaed1edfaab2dd2c4/etna/search/views.py#L416) in this PR 
The reason for removing this see my comment.
> In order to make the bucket attribute available before the API call, it was required to `set_current_bucket` when group value was given.

Reply:
@ababic  Earlier it seemed confusing to know that `get_buckets()` also sets the current bucket. We need to access `self.bucket_list.current.aggregations` before the earlier setting  `is_current` status in `get_buckets()` which is called via `get_context_data` . Therefore `set_current_bucket` does that before `get_context_data`.

Reply:
What's the best way forward here, @TNA-Allan?

Reply:
>  Now we have two methods for preparing that data that need to be called in a specific order. Surely we can find a way to make this work that doesn't add this unnecessary complication?

Yes 
1) set_current_bucket() - set the currents bucket after the form is validated which is available before calling the API and required so that we use `self.bucket_list.current.aggregations` before the API call.
2) get_buckets() - uses data after the API is called which serves other purposes  Ex: update result_count which is available only after API call.

Its following a general approach to way/pattern Bucket attributes are initialised. I don't think its complicated.

Reply:
@ababic I will try using BucketList.get_bucket()

Reply:
> Hi @TNA-Allan. I have implemented the get_current_bucket() changes in a separate PR, as I think that's where most of the confusion has come from: https://github.com/nationalarchives/ds-wagtail/pull/811

Thanks for this, will apply changes from PR once we're ready.

Reply:
PR [811]( https://github.com/nationalarchives/ds-wagtail/pull/811)  applied